### ISSUE TITLE
One of the game windows close after switching back and forth between window

### ISSUE THREAD

**Issue by mbb3-mitch** [shupershuff/Diablo2RLoader#33]
I followed setup procedure as follows:

1. Before trying this script I followed the simple process explorer routine, without success.
2. While trying to debug the issue:

- I went through the readme and went through all the configurations in the xml and csv files.
- I run the script from an administrator VS code.
- I enabled multiple instances on battle net.

The script continues to run and I can reopen the closed account. The error doesn't usually occur on the first switching of windows, it's usually when I try to bring the characters to the same place or interact. I didn't find any log files or uncaught exceptions.

Basically, I just want to give my HC sorc some BO. :P

Let me know if I can provide you with anymore details.

![image](https://github.com/shupershuff/Diablo2RLoader/assets/18585131/ae8a0ca4-eeea-4fa7-84e0-778791f2be02)

**Comment by shupershuff** [comment]
Firstly and most importantly, you mentioned you run the script from Visual Studio instead of PowerShell. Have you tried launching it from the desktop icon instead?

If that doesn't fix it, a few more questions:
1. Does the script show any error message when this happens?
2. When you say primary account closes, does the Window itself fully close (leaving only one instance running) OR do both instances stay open with one instance reverting to the single player screen?
3. Have you placed handle.exe in the appropriate directory?
4. Are you using any mods?

**Comment by shupershuff** [comment]
> 1. Before trying this script I followed the simple process explorer routine, without success.
You mention you didn't get much success with old method. Did you experience the same behaviour here with one instance randomly shutting down?

It's certainly and odd issue but it's possible there isn't enough RAM and/or VRAM to run an additional instance and the extra instance is crashing.
Can you please list your PC specs?

**Comment by mbb3-mitch** [comment]
I did try the old method with process explorer and it did not work. Same behaviour I currently am experiencing.

I use full screen mode. 

Here are my specs:

Processor	11th Gen Intel(R) Core(TM) i7-11700F @ 2.50GHz   2.50 GHz
Installed RAM	16.0 GB (15.7 GB usable)
System type	64-bit operating system, x64-based processor
GPU                 NVIDIA GeForce RTX 3070 8GB
OS                    Windows 11

![image](https://github.com/shupershuff/Diablo2RLoader/assets/18585131/e1ab2249-9458-4c8a-ad7f-854ba6cbee8d)


I tried optimizing all my settings for performance. Making windows 1080p, ultra performance, low graphics settings.

I managed to BO and teleport, but then switching back to barb, it closed my other window.

**Comment by mbb3-mitch** [comment]
@shupershuff 
![image](https://github.com/shupershuff/Diablo2RLoader/assets/18585131/823836ea-90ff-4b6b-82a2-f04b9553ad5a)

I commented out the lines that clear the host to keep track of the handles being killed. So I know that the script that handled killing check instances works.

**Comment by shupershuff** [comment]
What is your RAM utilisation with one instance running?

**Comment by shupershuff** [comment]
Have you had much luck with this since? It kind of seems like you're maxing out on memory usage and it's causing an instance to crash.
What other apps are running that can be closed? Have you tried just running in PowerShell instead of Visual Studio (it might reduce some RAM usage).

Also have you set a frame rate cap within the games video settings?
on that note, did you try changing the settings to that the game is in full screen (windowed) instead of full screen?

Some other thoughts, check that your GPU drivers are up to date and perhaps check if there is anything in Event Viewer that would suggest the cause of the crash.

**Comment by mbb3-mitch** [comment]
Hey @shupershuff 
Sorry for the delayed response. I think you might be right. I have not had any success getting my ram seems to alway be capped when I am playing with both windows. I have mostly given up at this point. I would think that 16gb would be more than enough ram, but it always appears to be at around 16gb of usage. 

I had some success make even smaller windows and capping the frames at 30fps, I can manage to go for about a minute before it crashes. With this window size, it's unplayable. I have a 49 in ultra wide, not sure if the massive monitor is what causes such high ram usage.

Drivers, OS, and bios are all up to date.

**Comment by shupershuff** [comment]
I think RAM is possibly your issue, if you can close any memory hogging apps prior to running you might have better luck.
I recently just upgraded from 32GB to 64GB as with 3 instances I would start maxing out. I could run 4 instances on 32GB but had similar issues where after a while a client would stutter or just crash. 16GB isn't much by today's standards so perhaps it's time for an upgrade :)

I don't believe larger resolution would impact RAM usage, but it will impact on VRAM (which can also cause crashes when it's maxed out).

**Comment by mbb3-mitch** [comment]
@shupershuff 
Just received some new ram. With 2 instances, it uses > 25gb of ram consistently.
